platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/rest/ApplicationUploadEndpoint.java

```diff
@@ -79,55 +82,50 @@ public Response update(MultipartBody multipartBody, @Context UriInfo requestUriI
     }
 
     try {
-      if (newFile != null) {
-        ZipFileApplicationDetails appDetails = ApplicationFileInstaller.getAppDetails(newFile);
-
-        if (appDetails != null) {
-          // lets get the existing app if it exists.
-          Application existingApplication = appService.getApplication(appDetails.getName());
-          boolean wasExistingAppStarted = false; // assume false until proved
-          // otherwise.
-          if (existingApplication != null) {
-            wasExistingAppStarted = appService.isApplicationStarted(existingApplication);
-            appService.removeApplication(existingApplication);
-          }
-          appService.addApplication(newFile.toURI());
-
-          // if application was started before it was removed, lets try and start it.
-          if (wasExistingAppStarted) {
-            appService.startApplication(appDetails.getName());
-          }
-        } else {
-          throw new ApplicationServiceException(
-              "No Application details could be extracted from the provided file.");
-        }
-      } else {
+      if (newFile == null) {
         throw new ApplicationServiceException("No file attachment provided.");
       }
 
-      // we need to output valid JSON to the client so fileupload can correctly call
-      // done/fail callbacks correctly.
+      ZipFileApplicationDetails appDetails = ApplicationFileInstaller.getAppDetails(newFile);
+
+      if (appDetails == null) {
+        throw new ApplicationServiceException(
+            "No Application details could be extracted from the provided file.");
+      }
+
+      Application existingApplication = appService.getApplication(appDetails.getName());
+      boolean wasExistingAppStarted = false;
+
+      if (existingApplication != null) {
+        wasExistingAppStarted = appService.isApplicationStarted(existingApplication);
+        appService.removeApplication(existingApplication);
+      }
+      appService.addApplication(newFile.toURI());
+
+      // If application was started before it was removed, try and start it
+      if (wasExistingAppStarted) {
+        appService.startApplication(appDetails.getName());
+      }
+
+      // Output valid JSON to the client so fileupload can correctly call done/fail callbacks
+      // correctly.
       Response.ResponseBuilder responseBuilder =
           Response.ok("{\"status\":\"success\"}").type("application/json");
       response = responseBuilder.build();
+
     } catch (ApplicationServiceException e) {
       LOGGER.warn("Unable to update an application on the server: {}", newFile, e);
       Response.ResponseBuilder responseBuilder = Response.serverError();
       response = responseBuilder.build();
     }
 
-    LOGGER.trace("EXITING: update");
```

Comment:
Thank you, I hate these traces. They are confusing as hell since they don't trace exit from all possible paths.

❓ Based on what you did below, didn't you mean to also remove the entry traces at the beginning of this method?